narwhal/primary/src/primary.rs

```diff
@@ -464,6 +330,22 @@ impl Primary {
             node_metrics,
         );
 
+        // spawn a task to swap the controllers in network when shutdown
+        let mut rx_reconfigure = tx_reconfigure.subscribe();
+        let shutdown_monitor_handle = spawn_monitored_task!(async move {
+            while (rx_reconfigure.changed().await).is_ok() {
+                let message = rx_reconfigure.borrow().clone();
+                if let ReconfigureNotification::Shutdown = message {
+                    // swap the handlers
+                    primary_receiver_controller.swap(None);
+                    worker_receiver_controller.swap(None);
```

Comment:
We want to swap the controllers with `None` once the `Shutdown` signal is received so our endpoints can start responding with the `500 Service not ready` message (another more descriptive could be used with more appropriate code)